esp/esdllib/esdl_def_helper.cpp

```diff
@@ -119,15 +119,36 @@ void removeEclHiddenStructs(IPropertyTree &depTree)
     ForEach(*it)
         depTree.removeTree(&it->query());
 }
-void removeEclHiddenElements(IPropertyTree &depTree)
+void removeEclHiddenElements(IPropertyTree &depTree, bool cloneGetDataFrom)
 {
     Owned<IPropertyTreeIterator> it = depTree.getElements("*");
     ForEach(*it)
     {
         StringArray names;
-        Owned<IPropertyTreeIterator> elements = it->query().getElements("*[@get_data_from]");
+        IPropertyTree &depChild = it->query();
+        Owned<IPropertyTreeIterator> elements = depChild.getElements("*[@get_data_from]");
         ForEach(*elements)
-            names.appendUniq(elements->query().queryProp("@name"));
+        {
+            IPropertyTree &element = elements->query();
+            StringAttr name = element.queryProp("@name");
+            if (!cloneGetDataFrom)
+                names.appendUniq(name);
```

Comment:
Should this be done for both cases?

Reply:
no, names is actually a very destructive array of item names.  They are to be deleted.  That's not what I want for these transformed items.   But it does make the point that I could change the array name to something more descriptive.